File: framework/fit/java/fit-builtin/plugins/fit-http-openapi3-swagger/src/main/java/com/huawei/fit/http/openapi3/swagger/Serializable.java
/*
 * Copyright (c) Huawei Technologies Co., Ltd. 2023-2023. All rights reserved.
 */

package com.huawei.fit.http.openapi3.swagger;

import java.util.Map;

/**
 * 表示可序列化为 JSON 对象的接口。
 *
 * @author 季聿阶
 * @since 2023-08-23
 */
public interface Serializable {
    /**
     * 将当前对象转换成 JSON 格式。
     *
     * @return 表示 JSON 格式的 {@link Map}{@code <}{@link String}{@code , }{@link Object}{@code >}。
     */
    Map<String, Object> toJson();
}

File: framework/fel/python/plugins/fel_langchain_loader_tools/types/serializable.py
# -- encoding: utf-8 --
# Copyright (c) Huawei Technologies Co., Ltd. -. All rights reserved.
import typing

try:
    import pydantic

    if pydantic.__version__.startswith("1."):
        raise ImportError
    import pydantic.v1 as pydantic
except ImportError:
    import pydantic


class Serializable(pydantic.BaseModel):
    def json(self, **kwargs: typing.Any) -> str:
        kwargs_with_defaults: typing.Any = {"by_alias": True, "exclude_unset": True, **kwargs}
        return super().json(**kwargs_with_defaults)

    def dict(self, **kwargs: typing.Any) -> typing.Dict[str, typing.Any]:
        kwargs_with_defaults: typing.Any = {"by_alias": True, "exclude_unset": True, **kwargs}
        return super().dict(**kwargs_with_defaults)